File: MedArchon.Web/App/Task/TaskViewModel.js
function TaskViewModel() {
	var self = this;

	self.Id = {};
	self.Description = {};
	self.Name = {};
    self.DueDate = ko.observable();
	self.Notes = ko.observableArray();
    
	self.CompleteTask = function () {
		Global.Api.Task.Complete(self.Id, function () {
			toastr.info("Task has been marked as completed.<br>" + self.Name);
			Global.Routing.Redirect('/');
		});
	};

    self.AddNote = function(text) {
        //whatever
    };
}

File: MedArchon.Web.ViewModels/TaskViewModel.cs
using System;

namespace MedArchon.Web.ViewModels
{
    public class TaskViewModel
    {
        public Guid Id { get; set; }
        public string Description { get; set; }
        public DateTime DueDate { get; set; }
        public string Name { get; set; }
        public bool Complete { get; set; }
    }
}